Spencer-Damiano/Fawna
# Backend

# Implement MongoDB Connection and API Endpoints

## Description
This pull request implements a FastAPI backend with MongoDB integration, focusing on connecting to the `sample_mflix` database and retrieving sample data. It also includes improved error handling and logging for better debugging and monitoring.

## Changes Made

### 1. MongoDB Connection
- Implemented MongoDB connection using `pymongo`
- Added environment variable support for MongoDB URI using `python-dotenv`
- Created startup and shutdown events for managing the database connection

### 2. API Endpoints
- Implemented the following endpoints:
  - `/`: Root endpoint returning a "Hello World" message
  - `/test_db`: Endpoint to test the MongoDB connection
  - `/mflix_comments`: Endpoint to retrieve sample comments from the `sample_mflix` database

### 3. Error Handling and Logging
- Added comprehensive error handling for database operations
- Implemented detailed logging throughout the application for better debugging
- Used `HTTPException` for proper error responses in API endpoints

### 4. Configuration and Utilities
- Added `load_dotenv()` to load environment variables from a `.env` file
- Implemented JSON serialization for MongoDB documents using `bson.json_util`

### 5. Testing
- Included a separate `test_mongodb_connection.py` script for standalone testing of the MongoDB connection

## Implementation Details
- The application uses FastAPI for creating the web API
- MongoDB connection is established using `pymongo.MongoClient`
- The app connects specifically to the `sample_mflix` database
- Uvicorn is used as the ASGI server to run the FastAPI application

## How to Test
1. Ensure the `.env` file is properly configured with the MongoDB URI
2. Run the application using `python app.py`
3. Test the endpoints:
   - `GET /`: Should return a "Hello World" message
   - `GET /test_db`: Should confirm successful connection to MongoDB
   - `GET /mflix_comments`: Should return 5 sample comments from the database

## Notes
- Make sure to install the required dependencies listed in `requirements.txt`
- The MongoDB URI should be kept secret and not committed to the repository

This pull request lays the groundwork for further backend development, providing a solid foundation for database operations and API endpoints.
